go/app/main.go

```diff
@@ -20,22 +24,131 @@ type Response struct {
 	Message string `json:"message"`
 }
 
+type ResponseItems struct {
+	Items []ItemInfo `json:"items"`
+}
+
+type ItemInfo struct {
+	Name          string `json:"name"`
+	Category      string `json:"category"`
+	ImageFilename string `json:"image_filename"`
+}
+
 func root(c echo.Context) error {
 	res := Response{Message: "Hello, world!"}
 	return c.JSON(http.StatusOK, res)
 }
 
 func addItem(c echo.Context) error {
+	// Open and read items.json
+	var jsonItems ResponseItems
+	jsonFile, err := os.Open("items.json")
+	if err != nil {
+		fmt.Println("fail to open items.json")
+	}
+	defer jsonFile.Close()
+	jsonData, err := ioutil.ReadAll(jsonFile)
+	//convert []byte to json
+	json.Unmarshal(jsonData, &jsonItems)
+
 	// Get form data
 	name := c.FormValue("name")
 	c.Logger().Infof("Receive item: %s", name)
+	category := c.FormValue("category")
+	imgFile, err := c.FormFile("image")
+	var imgFileName string
+	if err != nil {
+		fmt.Println("fail to open image file")
+	} else {
+		// open the image file
+		imgData, err := imgFile.Open()
+
+		if err != nil {
+			fmt.Println("fail to read image file")
+			return err
+		}
+		defer imgData.Close()
+
+		bytes, _ := ioutil.ReadAll(imgData)
+
+		// sha256
+		ShaInstance := sha256.New()
+		ShaInstance.Write([]byte(bytes))
+		shaRes := ShaInstance.Sum(nil)
+		fmt.Printf("%x\n\n", shaRes)
+		imgFileName = hex.EncodeToString(shaRes) + ".jpg"
+
+		// save the image file with hashed name
+		os.WriteFile(imgFileName, bytes, 0666)
+	}
+
+	itemData := &ItemInfo{
+		Name:          name,
+		Category:      category,
+		ImageFilename: imgFileName,
+	}
+
+	jsonItems.Items = append(jsonItems.Items, *itemData)
+
+	// convert json to []byte and write it in json file
+	data, _ := json.MarshalIndent(jsonItems, "", "\t")
+	os.WriteFile("items.json", data, 0777)
 
 	message := fmt.Sprintf("item received: %s", name)
 	res := Response{Message: message}
-
 	return c.JSON(http.StatusOK, res)
 }
 
+func getItems(c echo.Context) error {
+	jsonFile, err := os.Open("items.json")
+	if err != nil {
+		fmt.Println("error opening json file")
+		return err
+	}
+
+	defer jsonFile.Close()
+
+	jsonData, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		fmt.Println("fail to read json file")
+		return err
+	}
+
+	var jsonContent ResponseItems
+	json.Unmarshal(jsonData, &jsonContent)
+	return c.JSON(http.StatusOK, jsonContent)
+
+}
+
+func getItemsDetail(c echo.Context) error {
+	itemId := c.Param("itemId")
+	// string to int
+	id, err := strconv.Atoi(itemId)
+
+	jsonFile, err := os.Open("items.json")
+	if err != nil {
+		fmt.Println("error opening json file")
+		return err
+	}
+	defer jsonFile.Close()
+
+	jsonData, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		fmt.Println("fail to read json file")
+		return err
+	}
+
+	var jsonContent ResponseItems
+	json.Unmarshal(jsonData, &jsonContent)
+	if id-1 >= 0 && id-1 < len(jsonContent.Items) {
```

Comment:
In future we will get rid of this logic and just use `id` to fetch record by id from the DB

Reply:
(no actions required now)

Reply:
> In future we will get rid of this logic and just use `id` to fetch record by id from the DB

Ok. Thanks for your review 😀.